tests/end_to_end_cases/scenario.rs

```diff
@@ -611,6 +611,24 @@ pub async fn fixture_broken_catalog(db_name: &str) -> ServerFixture {
     let mut path = fixture.dir().to_path_buf();
     path.push(server_id.to_string());
     path.push(db_name);
+    let mut generations: Vec<_> = std::fs::read_dir(path.clone())
+        .unwrap()
+        .flatten()
+        .filter_map(|entry| {
+            if entry.path().is_dir() {
+                Some(entry.path())
+            } else {
+                None
+            }
+        })
```

Comment:
```suggestion
        .filter_map(|entry| {
        	let path = entry.path();
            if path.is_dir() {
                Some(path)
            } else {
                None
            }
        })
```

Minor, but this was allocating twice.

You could also be new-Rust-fancy and do

```suggestion
        .filter_map(|entry| {
        	let path = entry.path();
            path.is_dir().then(|| path)
        })
```